### ISSUE TITLE
[Feature] Elaborate release notes on how Redis licensing leads to removal in `4.1.0` / allow OS alternatives

### ISSUE THREAD

**Issue by ToshY** [AmruthPillai/Reactive-Resume#1894]
### Is there an existing issue for this feature?

- [X] Yes, I have searched the existing issues and it doesn't exist.

### Feature Description

I've noticed from the release notes that a minor release [4.1.0](https://github.com/AmruthPillai/Reactive-Resume/releases/tag/v4.1.0) denotes the removal of Redis from using in Reactive Resume. 

I'm familiar that Redis changed it's license, but not the actual impact or the restrictions that come with it, and therefore I would like to know which part of the license changes actually impact Reactive Resume and lead to the decision to remove it entirely from the stack. 

After reading the [Redis blog](https://redis.io/blog/redis-adopts-dual-source-available-licensing/), I've found the following in the FAQ:

> 6. Who is impacted by this change?
>
> Organizations providing competitive offerings to Redis will no longer be permitted to use new versions of the source code of Redis free of charge under either of the dual licenses. Commercial licensing terms are available and can enable use cases beyond the RSALv2 or SSPLv1 license limitations. If you are building a solution that leverages Redis, but does not specifically compete with Redis itself, there is no impact. 

> 18. Can I continue to provide professional services around Redis products?
>
> Yes. We have a large ecosystem of systems integrator partners that provide consulting and professional services to help users deploy, manage, and operate our products for their internal use. The change to our license is not intended to deter partners from providing those services, and we will continue to encourage and support these types of systems integrator partners. Instead, the RSALv2 simply prevents embedding or hosting our community products in a manner competitive with ours.

From the above I can take away 2 main points:

1. _"If you are building a solution that leverages Redis, but does not specifically compete with Redis itself, there is no impact."_
2. _"Instead, the RSALv2 simply prevents embedding or hosting our community products in a manner competitive with ours."_

As Reactive Resume only leverages Redis (for caching etc.) and does not provide or host it in a competitive manner, why was it removed?

---

TLDR;

- Elaborate on removal of Redis.
- Allow back Redis (or OS variant <7.4) or allow alternatives (e.g. Dragonfly, Valkey, KeyDB).

**Comment by AmruthPillai** [comment]
The major reason why Redis was removed was:
1. There was very little use for caching data when there are so many updates made to a resume at once, that most cached data becomes invalidated instantly.
2. What was stored using Redis (the views/download counter for each resume) could have easily been stored in the main database (Postgres) instead, thus removing a whole lot of packages and dependencies from the server side.

Redis was initially embedded into the project to cache resume data and try to optimize the workflow, but it was later removed because it was causing a lot of inconsistencies in what the user sees (loading stale data over the current). You could chalk that up to me not knowing how to use Redis correctly.

The license change did come as a surprise and I didn't read too much into it, but just made sense that if the project isn't depending on Redis too much, might as well get rid of it. There isn't really a need to bring back any sort of key value store DB into the project when there's already a database that can be utilized.
